### DISCUSSION TITLE
can we support override strict mode of esm module without using plugin

### DISCUSSION THREAD

**Discussion by hardfist** [webpack/webpack#18367] [Q&A]
we want to convert esm to non-strict mode so it could remove the iife wrapper of entry module entirely (the iife wrapper cause performance regression in  quickjs engine)
https://github.com/webpack/webpack/blob/c586c7b1e027e1d252d68b4372f08a9bce40d96c/lib/javascript/JavascriptModulesPlugin.js#L839
we can modify strict only plugin now, and I'm wondering whether we could support it in configuration like
```js
module.exports = {
  //...
  module: {
    rules: [
      {
        //...
        parser: {
          strict: false, // **override strict mode**
          amd: false, // disable AMD
        },
      },
    ],
  },
};
```

**Comment by alexander-akait** [comment]
Sorry for delay, a lot of issues, yeah, feel free to send a PR, I am fine with it
